enhancements/declarative-index-config.md

```diff
@@ -931,12 +978,12 @@ Since the new `opm` binary will create configs on `add`, the new images will be
 
 ```Dockerfile
 ENTRYPOINT ["/bin/opm"]
-CMD ["registry", "serve", "--config", "config.json"]
+CMD ["registry", "serve", "config.json"]
 ```
 
 To migrate existing index images built with the old configuration, all existing `opm index` commands that alter the index (i,e `opm index add --from-index`, `opm index prune`, `opm index prune-stranded` and `opm index rm`) will be enhanced to first let the corresponding operation through, and then check if the sqllite database exists in the index. If it does, the sqllite database will be converted over to package representations instead, the sqllite database will be deleted, and the new altered image will be built using the new configuration. `prune`, `prune-stranded` and `rm` sub-commands under the `opm index` command will also be marked for deprecation. If these sub-commands are used on an index that has already been migrated over to package representations, the operations will still be supported on these indexes with package representations for the period of time these sub-commands are marked as deprecated, until they are removed.      
```

Comment:
is there a simple way or tool envisioned to only convert an existing index.db to the json representation?  without having to add to the index or remove.

Reply:
Yes - I believe there will be a `migrate` command to do exactly that. @anik120 may have more details.

There's currently a POC for this in https://github.com/joelanford/operator-registry/tree/feat/serve-db-via-model. If you run `opm registry serve my-index.db`, it'll start up the usual GRPC server AND write out a `my-index-configs` directory with the declarative config representation of the DB.

Reply:
>  If you run opm registry serve my-index.db, it'll start up the usual GRPC server AND write out a my-index-configs directory with the declarative config representation of the DB.

do we have a feel for whether an implicit conversion like this provides value? I'm thinking that an explicit db -> config conversion command would be more straightforward.

Reply:
@njhale Totally agree. I just have that branch setup that way to make it quick and easy to test out the declarative config representation of existing DBs. I don't think we want `opm registry serve` to do migrations.